### PR TITLE
Update `parking-management` solution

### PR DESCRIPTION
**Pull request by RizwanMunawar**
@glenn-jocher, This PR is good to merge 👍🏻 

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
This update enhances the parking management solution by improving the Tkinter UI and refactoring the code for better functionality and maintainability.

### 📊 Key Changes
- Added `json_file` parameter to the configuration for parking system regions.
- Simplified the `ParkingPtsSelection` class by reorganizing code and refining UI setup processes.
- Refactored the `ParkingManagement` class with enhancements for JSON data handling and visualization colors.
- Updated import statements and removed unnecessary ones for improved readability and efficiency.
- Enhanced image handling to ensure better UI responsiveness and usability in the Tkinter interface.

### 🎯 Purpose & Impact
- **Improved User Interface**: The refactored Tkinter setup makes the UI more intuitive and organized, benefiting users selecting parking zone points.
- **Better Code Maintenance**: Streamlined code structure aids in easier maintenance and future enhancements, impacting developers positively.
- **Enhanced User Experience**: Simplified image upload and handling allow users to manage bounding boxes more efficiently, leading to a smoother experience.
- **Accurate Parking Management**: By utilizing the `json_file` for region definitions, parking occupancy monitoring becomes more precise, ultimately improving real-time parking management capabilities.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @RizwanMunawar, thank you for submitting a PR to the `ultralytics/ultralytics` 🚀 repository! This is an automated response to guide you through the review process. One of our engineers will be with you shortly to assist further. In the meantime, please ensure the following:

- ✅ **Define a Purpose**: Clearly explain the purpose of your update to the `parking-management` solution in your PR description. If any issues are related, consider linking to relevant [issues page](https://github.com/ultralytics/ultralytics/issues) for better context.
- ✅ **Synchronize with Source**: Make sure your PR is up to date with the `main` branch of the `ultralytics` repository. If it's not, you can update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: All Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks should be passing. If any are failing, please investigate and resolve these issues.
- ✅ **Update Documentation**: Make any necessary changes to the relevant [documentation](https://docs.ultralytics.com) to reflect the updates in your PR.
- ✅ **Add Tests**: Where applicable, ensure there are tests that cover your changes. Confirm that all tests are passing successfully.
- ✅ **Sign the CLA**: If this is your first contribution to Ultralytics, please sign our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) by leaving a comment saying "I have read the CLA Document and I sign the CLA".
- ✅ **Minimize Changes**: As per best practices, ensure your changes are limited to what is necessary for the PR. _"Focus is saying no to something else." — Steve Jobs_

Please let us know if you require further assistance or have any questions. While we process your PR, you might find our [Contributing Guide](https://docs.ultralytics.com/help/contributing) helpful. Thank you for your efforts and contribution to the Ultralytics community! 🚀

**Comment by glenn-jocher** [comment]
@RizwanMunawar merged!

File: ultralytics/solutions/parking_management.py [hunk 1]
```diff
@@ -4,238 +4,168 @@
...
+        self.image = self.canvas_image = None
+        self.rg_data, self.current_box = [], []
+        self.imgw = self.imgh = 0
+        self.canvas_max_width, self.canvas_max_height = 1280, 720
 
     def upload_image(self):
-        """Upload an image and resize it to fit canvas."""
-        from tkinter import filedialog
-
-        from PIL import Image, ImageTk  # scope because ImageTk requires tkinter package
```
Comment (glenn-jocher) [inline ultralytics/solutions/parking_management.py:60]: @RizwanMunawar I had this here for a reason and after you removed it now our Docker images are crashing.

Comment (RizwanMunawar) [reply ultralytics/solutions/parking_management.py:60]: @glenn-jocher opened the PR for this fix: https://github.com/ultralytics/ultralytics/pull/16995

**Pull request merged**